src/main/java/com/bravo/user/dao/specification/UserSpecification.java

```diff
@@ -4,7 +4,9 @@
 import com.bravo.user.model.filter.UserFilter;
 import javax.persistence.criteria.CriteriaBuilder;
 import javax.persistence.criteria.CriteriaQuery;
+import javax.persistence.criteria.Path;
 import javax.persistence.criteria.Root;
+import java.util.Set;
 
 public class UserSpecification extends AbstractSpecification<User> {
```

Comment:
Similar to `AbstractSpecification` comment. This class is only used for searching and invoked from the service layer. The validation should happen before the code reaches this point.